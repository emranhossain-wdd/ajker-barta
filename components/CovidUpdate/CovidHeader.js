import Link from 'next/link';
import React, { useEffect, useState } from 'react';
import styles from '../../styles/Covid.module.css'

const CovidHeader = () => {
  const [timeSeries, setTimeSeries] = useState('')
  useEffect(() => {
    fetch('https://api.covid19api.com/summary')
      .then(res => res.json())
      .then(data => setTimeSeries(data?.Date))
  }, [])
  return (
    <div className={styles.header}>
      <div className="container">
        <div className='sm:flex items-center justify-between'>
          <div className={styles.logo}>
            <img src='https://i.postimg.cc/2jbfZ5hW/coronavirus.png' alt="" />
            <h1>Covid Update</h1>
          </div>
          <p>{`${timeSeries?.split('T')[0]} at ${timeSeries?.split('T')[1]}`}</p>
          <Link href='/'><a>Back To Home</a></Link>
        </div>
      </div>
    </div >
  );
};

export default CovidHeader;