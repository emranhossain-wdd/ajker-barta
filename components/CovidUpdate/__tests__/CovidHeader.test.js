import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CovidHeader from '../CovidHeader'

it('render CovidHeader component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<CovidHeader />, div)
})
it('maching snapshot', () => {
  const tree = renderer.create(<CovidHeader />)
  expect(tree).toMatchSnapshot()
})