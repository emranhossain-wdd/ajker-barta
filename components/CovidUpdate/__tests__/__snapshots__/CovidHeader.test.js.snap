// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <div
      className="sm:flex items-center justify-between"
    >
      <div
        className="logo"
      >
        <img
          alt=""
          src="https://i.postimg.cc/2jbfZ5hW/coronavirus.png"
        />
        <h1>
          Covid Update
        </h1>
      </div>
      <p>
         at undefined
      </p>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Back To Home
      </a>
    </div>
  </div>
</div>
`;
